astropy/coordinates/attributes.py

```diff
@@ -92,39 +92,64 @@ def convert_input(self, value):
         """
         return value, False
 
+    def validate(self, instance):
+        """
+        Validate the current value using `convert_input` and update the
+        value with the validated result if it changed.
+        """
+        name = f"_{self.name}"
+        value = getattr(instance, name)
+
+        # we do not change / validate None here
+        if value is None:
+            return
+
+        value, changed = self.convert_input(value)
+        if changed:
+            setattr(instance, name, value)
+
+        return value
+
+    def broadcast(self, instance):
+        """
+        Apply broadcasting to this attribute for the given instance.
+        """
+
+        name = f"_{self.name}"
+        out = getattr(instance, name)
+        out_shape = getattr(out, 'shape', ())
+
+        instance_shape = getattr(instance, 'shape', None)  # None if instance (frame) has no data!
+        converted = False
+        if instance_shape is not None and out_shape != () and out_shape != instance_shape:
+            # If the shapes do not match, try broadcasting.
+            try:
+                if isinstance(out, ShapedLikeNDArray):
+                    out = out._apply(np.broadcast_to, shape=instance_shape,
+                                     subok=True)
+                else:
+                    out = np.broadcast_to(out, instance_shape, subok=True)
+            except ValueError:
+                # raise more informative exception.
+                raise ValueError(
+                    "attribute {} should be scalar or have shape {}, "
+                    "but is has shape {} and could not be broadcast."
+                    .format(self.name, instance_shape, out.shape))
+
+            converted = True
```

Comment:
Can just do `setattr(instance, name, out)` here - it is the only place where `converted` gets set (unlike before, when `convert_input` could set it).